cfme/tests/automate/test_domain_priority.py

```diff
@@ -38,6 +38,14 @@ def copy_domain(request):
     return domain
 
 
+@pytest.fixture(scope="module")
+def original_domain(request):
+    domain = Domain(name=fauxfactory.gen_alphanumeric(), enabled=True)
+    domain.create()
+    request.addfinalizer(lambda: domain.delete() if domain.exists() else None)
```

Comment:
yeh, no problem. I'd propose something like this if you decide to make it a yield_fixture:

```python
domain = Domain(...)
try:
    domain.create()
    yield domain
finally:
    if domain.exists:  # I think this property is there
        domain.delete()
```